spec/mixlib/install/backend_spec.rb

```diff
@@ -0,0 +1,216 @@
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages.s3.amazonaws.com/mac_os_x/10.7/x86_64/chef-12.2',
+              version: '12.2'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with a major product version' do
+          let(:product_version) { '12' }
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages.s3.amazonaws.com/mac_os_x/10.7/x86_64/chef-12',
+              version: '12'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with latest version keyword' do
+          let(:product_version) { 'latest' }
+          let(:expected_info) { {} }
+
+          it_behaves_like 'the right artifact info'
+        end
+      end
+    end
+
+    context 'for current' do
+      let(:channel) { :current }
+
+      context 'when p, pv and m are present' do
+        let(:platform) { 'mac_os_x' }
+        let(:platform_version) { '10.9' }
+        let(:architecture) { 'x86_64' }
+
+        context 'with an integration product version' do
+          let(:product_version) { '12.4.3+20151006083011' }
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages-current.s3.amazonaws.com/mac_os_x/10.9/x86_64/chef-12.4.3%2B20151006083011-1.dmg',
+              md5: '8acf2ebbb4a8a89a33ccc8650f25243e',
+              sha256: '9c93e86ab531fd1fcb95c3e291d0aed333d7af6634c93bbfd92f0773b30d1657',
+              version: '12.4.3+20151006083011'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with a major.minor.patch version' do
+          let(:product_version) { '12.4.3' }
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages-current.s3.amazonaws.com/mac_os_x/10.9/x86_64/chef-12.4.3%2B',
+              version: '12.4.3+'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with a major.minor product version' do
+          let(:project_version) { '12.4' }
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages-current.s3.amazonaws.com/mac_os_x/10.9/x86_64/chef-12.4',
+              version: '12.4'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with a major product version' do
+          let(:project_version) { '12' }
+          let(:expected_info) {
+            {
+              url: 'http://opscode-omnibus-packages-current.s3.amazonaws.com/mac_os_x/10.9/x86_64/chef-12',
+              version: '12'
+            }
+          }
+
+          it_behaves_like 'the right artifact info'
+        end
+
+        context 'with latest version keyword' do
+          let(:project_version) { 'latest' }
```

Comment:
IMO since we support symbols in `channel` a user would probably expect us to support `:latest` in the version as well.